spring-cloud-stream-binder-kafka-core/src/main/java/org/springframework/cloud/stream/binder/kafka/provisioning/KafkaTopicProvisioner.java

```diff
@@ -58,18 +69,23 @@
 public class KafkaTopicProvisioner implements ProvisioningProvider<ExtendedConsumerProperties<KafkaConsumerProperties>,
 		ExtendedProducerProperties<KafkaProducerProperties>>, InitializingBean {
 
+	private static final int DEFAULT_OPERATION_TIMEOUT = 30;
+
 	private final Log logger = LogFactory.getLog(getClass());
 
 	private final KafkaBinderConfigurationProperties configurationProperties;
 
-	private final AdminUtilsOperation adminUtilsOperation;
+	private final AdminClient adminClient;
 
 	private RetryOperations metadataRetryOperations;
 
-	public KafkaTopicProvisioner(KafkaBinderConfigurationProperties kafkaBinderConfigurationProperties,
-								AdminUtilsOperation adminUtilsOperation) {
+	private int operationTimeout = DEFAULT_OPERATION_TIMEOUT;
+
+	public KafkaTopicProvisioner(KafkaBinderConfigurationProperties kafkaBinderConfigurationProperties) {
 		this.configurationProperties = kafkaBinderConfigurationProperties;
-		this.adminUtilsOperation = adminUtilsOperation;
+		Map<String, Object> adminConfigs = new HashMap<>();
+		adminConfigs.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaBinderConfigurationProperties.getKafkaConnectionString());
+		this.adminClient = AdminClient.create(adminConfigs);
```

Comment:
We should use boot`s `KafkaProperties` bean and `buildAdminProperties()` - then override with any binder-specific properties such as the broker address.

Reply:
We'll need to be careful to explain that boot's `failFast` property won't take effect (the binder will now always start even if there are failures provisioning - when https://github.com/spring-cloud/spring-cloud-stream/pull/1131 is applied.

Reply:
(Or we could make it fail by throwing an `IllegalStateException` if there's a failure and the flag is true.

Reply:
Addressed the first comment. Can't the second comment be addressed as a docs story? (or a separate PR for the 3rd comment).